### PR TITLE
[MLIR] Add conversion of `tf.leakyRelu` and `tf.leakyReluGrad` from t…

### PR DESCRIPTION
**Pull request by pashu123**
…f to mhlo.

This commits implements conversion of tf.leakyRelu and tf.leakyReluGrad
operations from tf to mhlo. The changes have been made as a part of
xla-legalize-tf pass.

Signed-off-by: Prashant Kumar <prashantk@polymagelabs.com>

### CODE REVIEW

File: tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc [hunk 1]
```diff
@@ -1350,6 +1350,87 @@ class ConvertBroadcastToOp : public OpRewritePattern<TF::BroadcastToOp> {
...
+  LogicalResult matchAndRewrite(TF::LeakyReluOp op,
+                                PatternRewriter& rewriter) const override {
+    float alpha = op.alpha().convertToFloat();
+    Value features = op.features();
+    auto featureType = features.getType().cast<RankedTensorType>();
+    ArrayRef<int64_t> featureShape = featureType.getShape();
+    auto eltType = featureType.getElementType();
+
+    auto alphaVal = rewriter.create<mhlo::ConstOp>(
+        op.getLoc(), rewriter.getFloatAttr(eltType, alpha));
```
Comment (bondhugula) [inline tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc:1367]: Nit: You can use save the location to a `loc` - it has four uses in all. Some of the statements below may fit on a line.

Comment (pashu123) [reply tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc]: Done. Thanks!

File: tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc [hunk 2]
```diff
@@ -1350,6 +1350,87 @@ class ConvertBroadcastToOp : public OpRewritePattern<TF::BroadcastToOp> {
...
+ public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(TF::LeakyReluOp op,
+                                PatternRewriter& rewriter) const override {
+    Location loc = op.getLoc();
+    float alpha = op.alpha().convertToFloat();
+    Value features = op.features();
+    auto featureType = features.getType().cast<RankedTensorType>();
+    ArrayRef<int64_t> featureShape = featureType.getShape();
+    auto eltType = featureType.getElementType();
```
Comment (bondhugula) [inline tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc:1365]: auto -> Type

File: tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc [hunk 3]
```diff
@@ -1350,6 +1350,87 @@ class ConvertBroadcastToOp : public OpRewritePattern<TF::BroadcastToOp> {
...
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(TF::LeakyReluGradOp op,
+                                PatternRewriter& rewriter) const override {
+    Location loc = op.getLoc();
+    float alpha = op.alpha().convertToFloat();
+    Value gradients = op.gradients();
+    Value features = op.features();
+    auto featureType = features.getType().cast<RankedTensorType>();
+    ArrayRef<int64_t> featureShape = featureType.getShape();
+    auto eltType = featureType.getElementType();
```
Comment (bondhugula) [inline tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc:1407]: auto -> Type

**Review by bondhugula — APPROVED** [review]
LGTM. Two minor comments.

**Review by bondhugula — APPROVED** [review]

**Review by joker-eph — APPROVED** [review]

**Pull request merged**